### ISSUE TITLE
[Pixi 6.3.0] DisplacementFilter stops updating when browser goes full screen

### ISSUE THREAD

**Issue by Friksel** [pixijs/pixijs#8276]
There is a sprite bound to a DisplacementFilter. The sprite moves a little on each frame to move the displacement. This works fine, also when scaling the browser window, but stops working when the browser goes into fullscreen mode (F11 on Ms Edge) and once in full screen the normal update-behaviour for the filter never gets working again until reloading the webpage.
The filter is still applied in full screen, but it's not updating to the moving sprite anymore.

Not sure why this isn't working. In earlier versions I worked with filters and never saw this issue. 

There is nothing to see in the console.log.

I tried the following without succes:

- Setting the filterArea on the filter like so (and update it on a window resize event). This doesn't make a difference;

``` javascript
this.displFlt.filterArea = this.rend.screen;
```

- Setting both the sprite as the filter to the stage instead of on the `pixi-viewport` as a test; it still stops working when going fullscreen.

- Searching on the issue here on github. I found this commit that looks like it's created to fix some fullscreen issue on filters, but it's not clear to me if this needs a different workflow for us and/or what that workflow should be / if there's something we need to do in order to support full screen.

https://github.com/pixijs/pixijs/pull/4682

- After seeing some info here: https://github.com/pixijs/pixijs/wiki/v4-Creating-Filters  I also tried setting `autofit` of the filter to `true` and `false`, but that didn't change anything either.

- Next to Chromium browsers, also tested in Firefox latest (99.0). Same issue

Conclusion: I have no clue why suddenly what works in non-fullscreen mode stops working in fullscreen mode. So far it feels like an issue.

### Expected Behavior

DisplacementFilter to keep updating to moving texture (= displacement map) when switching browser to fullscreen.

### Current Behavior

DisplacementFilter stops updating to moving texture (= displacement map) when switching browser to fullscreen.

### Possible Solution

No clue

### Steps to Reproduce

``` javascript
// ...

initFilter() {
  this.displSpr = Sprite.from(texture);
  this.displSpr.texture.baseTexture.wrapMode = WRAP_MODES.REPEAT;
  this.vp.addChild(this.displSpr); // pixi-viewport
  
  this.displFlt = new filters.DisplacementFilter(this.displSpr, 1);
  this.vp.filters = [this.displFlt];
}


// using pixi ticker to trigger onTick();
onTick(e) {
   this.displSpr.x += e;
   this.displSpr.y += e;

   this.rend.render(this.stage);
}
```

### Environment

- **`pixi.js` version**:  6.3.0
- **Browser & Version**: Ms Edge (Chromium) 100.0.1185.36 (also tested in Firefox latest (99.0) on Windows. same issue)
- **OS & Version**: Windows 10

**Comment by bigtimebuddy** [comment]
Does this happen with other filters when full-screened? Like BlurFilter?

**Comment by Friksel** [comment]
> Does this happen with other filters when full-screened? Like BlurFilter?

Not sure how you'd like me to test BlurFilter. The displacementFilter itself is still applied when in full screen, only it's not updated anymore by the displacement sprite. AFAIK there's no way to update a BlurFilter to a sprite and can't think of another filter that's using a sprite to animate, so not sure how you'd like me to test this.

**Comment by bigtimebuddy** [comment]
Can you provide a minimal reproduction? I have a lot of questions about what you're doing. Is it a canvas size limit, is it a texture size limit, what size screen are you using, how are you doing the fullscreen... a reproduction will clarify.

**Comment by Friksel** [comment]
@bigtimebuddy 

>what size screen are you using, how are you doing the fullscreen...

It's on a HD screen (1920x1080p) with resolution 1. Which is the same size as the stage and canvas it was tested on.

>Is it a canvas size limit, is it a texture size limit, what size screen are you using, how are you doing the fullscreen... a reproduction will clarify.

The canvas gets the same size as the body, so max 1920x1080p, res 1. The sprite used as a displacement texture is just 256 x 256 px, so very small. The way I'm doing fullscreen was already fully described in my opening post; as described there I'm using the built in functionality of the browser by pressing F11 on Ms Edge or Firefox.

>I have a lot of questions about what you're doing. 
>Can you provide a minimal reproduction?

Yes, everybody wants a reproduction for everything these days and add a 'needs information' tag to postbone responsibility to fix something as that's easy. I wonder what questions you could have tho with basically only 3 lines of code and basic functionality like adding a filter and animating it with a moving sprite. I even put the code in my opening post. To hopefully solve an issue in your lib.

**Comment by domis86** [comment]
> Yes, everybody wants a reproduction for everything these days and add a 'needs information' tag to postbone responsibility to fix something as that's easy. 

If its "that easy" case then surely it wont be a problem to create running code for example in https://www.pixiplayground.com/ - right?

Remember that its open source project. People here are not working for you. Its free software. So please consider being more humble next time - it will get you further :)

**Comment by bigtimebuddy** [comment]
First thanks for reporting, it is valuable when people report all issues they find.

A reproduction is super useful because my goal when people submit issues is to try to first triage if its because a user-error because they're doing something undefined, anti-pattern or just flat wrong--which happens a lot. I can't tell you how many times I've done exactly what you're suggesting, only to find that I can't reproduce, then a user provides some other valuable info. It takes time even with a snippet like you provided.

Also, reproductions are valuable for validating if a fix works or if something regressed a year from now. We just rerun it! Not everything in Pixi can be easily run with a unit test, especially DOM behaviors like this issue.

The moral of the story is that make a repro or don't. BUT we are going to prioritize looking at issues that have clear reproductions and where the submitter did their homework to isolate and repro the issue on their own (try other computers, other versions of Pixi, other hardware). So it will probably take longer to look at issues with less info or context where I need to reverse-engineer what's happening.

**Comment by Friksel** [comment]
@domis86

> If its "that easy" case then surely it wont be a problem to create running code for example in https://www.pixiplayground.com/ - right?

I understand your point. However, it takes a lot of time to create these demos as there's always something taking things longer, like today; first I tried creating a project in codepen, to find out we have to deal with corrs issues when loading textures. Than I tried to create the demo in pixiplayground to find out pixiplayground isn't upgraded to pixijs 6.3.0, but still runs 5.3.0 (as seen in the console). Eventhough in the setting of the editor it was clearly set to 6.3.0.

Than in the pixiplayground I found out that the filters part wasn't recognized, so couldn't add the DisplacementFilter. That's probably something I did wrong, or filters were implemented differently in 5.3.0 (as the editor wasn't on 6.3.0). But again it took a long time to just setup the environment. I'm still not sure how to even use filters in that editor.

Than I found that imgur don't have issues with Corrs, so that's great to know now. So switched back to codepen. And back to my project.

> Remember that its open source project. People here are not working for you. Its free software. So please consider being more humble next time - it will get you further :)

You're completely right in that it's an open source project and I'm completely aware of that. However I think people these days forget how valuable feedback is and how much time and money goes into that that the open source lib benefits from hugely. Also don't forget that, although I'm very grateful for pixijs and I think it's a wonderful project, we also shouldn't forget that it's also a great benefit for GoodBoy to get jobs by being a clear expert in the field, have lots of exposure and made a name for themselves with this. And basically we also improve the lib for them. Which, don't get me wrong, it totally earned and they deserve it and it also surely takes a lot of effort, time and money for them.

Also don't forget that it's practically impossible these days to do your job without using open source projects. There are no paid alternatives and when there would be you'd never get a client as you'll always be more expensive than the competition. So open source is great, but it also has it's cons.

So my point is; it's not all one sided. This is a two way cooperation we all benefit from.

That said, I completely agree with you on that 'being humble' is a lot better to achieve things.

@bigtimebuddy

> First thanks for reporting, it is valuable when people report all issues they find.
> A reproduction is super useful because my goal when people submit issues is to try to first triage if its because a user-error because they're doing something undefined, anti-pattern or just flat wrong--which happens a lot. I can't tell you how many times I've done exactly what you're suggesting, only to find that I can't reproduce, then a user provides some other valuable info. It takes time even with a snippet like you provided.

First of all I admire the way you reacted here and put this into words. You come across as a true professional. Also I understand that there are a lot of cases that it would take a lot of time for you to investigate.

I assumed wrongly that you didn't even look at the code, but according to your post you did. So sorry for that.

Also sorry for not diving in further. Today I finally got a little more time to dive deeper into this just to find out the reason why it's also a great way to create a stripped demo; to make sure it's not something else going on.
Just to find out it was; just found out the issue I had didn't have anything to do with the filter. 

First I isolated the issue by only moving the sprite used in the filter while not used in the filter, but only on the stage. To find out that already that stopped moving when going full screen. For moving I used the delta time of the ticker in `onTick(e)` but for some reason in the project there as an `onTick(e)` send when resizing the window too, which didn't have that delta parameter. So the sprite was trying to add `undefined` to the current x and y of the sprite positions.... Which isn't possible of course and caused the calculation to fail and keep failing afterwards when going full screen.

I'm very sorry for taking your time with this. And I should have dived deeper into this earlier. 

Also, I should have dimmed my tone and shouldn't have projected experiences with other libs unto pixijs. PixiJs is one of the libs that's very well maintained, by clearly passionate people who understand what they're doing and I'm always happy to use it.
Thanks again for the way you took this, my attituded wasn't fitting and in the end it was my own fault in my own code.

Never to old to learn!

Keep up the great work and hava a nice day!
